ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -140,33 +149,33 @@ public void run(DatabaseClient client, InputStream csvRecords, InputStream jobCo
                     	 if(getBatchSize()==1) {
                     		 System.out.println("Success for the batch-" + batch.getJobBatchNumber() + ". Total successes - 1");
                     	 } else {
-                    		 if(jobRunnerVariables.totalSuccessCount == 0){
+                    		 if(jobRunnerVariables.totalSuccessCount.equals(0)){
                     			 logger.info("Success for batch number - " + batch.getJobBatchNumber());
                     			 setLastSuccessLogTime(System.currentTimeMillis());
-                    			 jobRunnerVariables.lastSuccessCount = 1;
+                    			 jobRunnerVariables.lastSuccessCount.set(1);;
                     		 } else if(((System.currentTimeMillis() - getLastSuccessLogTime())/1000) >= getTimeIntervalInSeconds()) {
                     			 setLastSuccessLogTime(System.currentTimeMillis());
-                    			 logger.info("Success for batch number - " + batch.getJobBatchNumber() + ". Batch running since last log - " + (jobRunnerVariables.totalSuccessCount - jobRunnerVariables.lastSuccessCount + 1) + ". Total successes - " + jobRunnerVariables.totalSuccessCount + 1);
-                    			 jobRunnerVariables.lastSuccessCount = jobRunnerVariables.totalSuccessCount + 1;
+                    			 logger.info("Success for batch number - " + batch.getJobBatchNumber() + ". Batch running since last log - " + (jobRunnerVariables.totalSuccessCount.get() - jobRunnerVariables.lastSuccessCount.get() + 1) + ". Total successes - " + jobRunnerVariables.totalSuccessCount + 1);
+                    			 jobRunnerVariables.trackLastSuccessCount();
                     		 } 
```

Comment:
Looking carefully at lines 152 through 159, I can see that I didn't think through all of the opportunities for race conditions.

Really, only one thread should be able to enter a branch. 

What if the test and field assignments were moved into a synchronized method of jobRunnerVariables that returns true if the lambda should log?

Reply:
Unless there is some future need for extending the class, you could make jobRunnerVariables an enum so that thread safety is enforced at the jvm level

Reply:
Moved the else if block code to the synchronized block. Hope that helps.